File: app/src/main/java/tv/superawesome/demoapp/AdMobActivity.java
package tv.superawesome.demoapp;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.InterstitialAd;
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.reward.RewardItem;
import com.google.android.gms.ads.reward.RewardedVideoAd;
import com.google.android.gms.ads.reward.RewardedVideoAdListener;

import tv.superawesome.lib.sasession.SAConfiguration;
import tv.superawesome.plugins.publisher.admob.SAAdMobExtras;
import tv.superawesome.plugins.publisher.admob.SAAdMobBannerCustomEvent;
import tv.superawesome.plugins.publisher.admob.SAAdMobInterstitialCustomEvent;
import tv.superawesome.plugins.publisher.admob.SAAdMobVideoMediationAdapter;
import tv.superawesome.sdk.publisher.SAOrientation;

// app: ca-app-pub-7706302691807937~5001530003
// banner: ca-app-pub-7706302691807937/1989188001
// interstitial: ca-app-pub-7706302691807937/6478263208
// video: ca-app-pub-7706302691807937/3465921207

public class AdMobActivity extends Activity {

    private AdView adView;
    private InterstitialAd mInterstitialAd;
    private RewardedVideoAd mAd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_admob);

        MobileAds.initialize(this, "ca-app-pub-7706302691807937~5001530003");

        Bundle extras1 = SAAdMobExtras.extras()
                    .setTestMode(false)
                    .setConfiguration(SAConfiguration.STAGING)
                    .setParentalGate(false)
                    .setTransparent(true)
                    .build();

        adView = (AdView) findViewById(R.id.adView);
        adView.setAdListener(new AdListener() {
            @Override
            public void onAdClosed() {
                Log.d("SADefaults/AdMob", "Banner ad closed");
            }

            @Override
            public void onAdFailedToLoad(int i) {
                Log.d("SADefaults/AdMob", "Banner ad failed to load");
            }

            @Override
            public void onAdLeftApplication() {
                Log.d("SADefaults/AdMob", "Banner ad left application");
            }

            @Override
            public void onAdOpened() {
                Log.d("SADefaults/AdMob", "Banner ad opened");
            }

            @Override
            public void onAdLoaded() {
                Log.d("SADefaults/AdMob", "Banner ad loaded");
            }
        });
        adView.loadAd(new AdRequest.Builder().addCustomEventExtrasBundle(SAAdMobBannerCustomEvent.class, extras1).build());


        Bundle extras2 = SAAdMobExtras.extras()
                .setTestMode(false)
                .setConfiguration(SAConfiguration.STAGING)
                .setOrientation(SAOrientation.PORTRAIT)
                .setParentalGate(true)
                .build();

        mInterstitialAd = new InterstitialAd(this);
        mInterstitialAd.setAdUnitId("ca-app-pub-7706302691807937/6478263208");
        mInterstitialAd.setAdListener(new AdListener() {
            @Override
            public void onAdClosed() {
                Log.d("SADefaults/AdMob", "Interstitial ad closed");
            }

            @Override
            public void onAdFailedToLoad(int i) {
                Log.d("SADefaults/AdMob", "Interstitial ad failed to load");
            }

            @Override
            public void onAdLeftApplication() {
                Log.d("SADefaults/AdMob", "Interstitial ad left application");
            }

            @Override
            public void onAdOpened() {
                Log.d("SADefaults/AdMob", "Interstitial ad opened");
            }

            @Override
            public void onAdLoaded() {
                Log.d("SADefaults/AdMob", "Interstitial ad loaded");
            }
        });
        mInterstitialAd.loadAd(new AdRequest.Builder().addCustomEventExtrasBundle(SAAdMobInterstitialCustomEvent.class, extras2).build());

        mAd = MobileAds.getRewardedVideoAdInstance(this);

        mAd.setRewardedVideoAdListener(new RewardedVideoAdListener() {
            @Override
            public void onRewardedVideoAdLoaded() {
                Log.d("SADefaults/AdMob", "Video Ad Loaded");
            }

            @Override
            public void onRewardedVideoAdOpened() {
                Log.d("SADefaults/AdMob", "Video Ad opened");
            }

            @Override
            public void onRewardedVideoStarted() {
                Log.d("SADefaults/AdMob", "Video Ad Started");
            }

            @Override
            public void onRewardedVideoAdClosed() {
                Log.d("SADefaults/AdMob", "Video AD Closed");
            }

            @Override
            public void onRewarded(RewardItem rewardItem) {
                Log.d("SADefaults/AdMob", "Video Ad Rewarded");
            }

            @Override
            public void onRewardedVideoAdLeftApplication() {
                Log.d("SADefaults/AdMob", "Video Ad Left app");
            }

            @Override
            public void onRewardedVideoAdFailedToLoad(int i) {
                Log.d("SADefaults/AdMob", "Video Ad Failed to load");
            }

            @Override
            public void onRewardedVideoCompleted() {
                Log.d("SADefaults/AdMob", "Video Ad Completed");
            }
        });

        Bundle extras3 = SAAdMobExtras.extras()
                .setTestMode(false)
                .setConfiguration(SAConfiguration.STAGING)
                .setParentalGate(false)
                .setOrientation(SAOrientation.LANDSCAPE)
                .setSmallClick(true)
                .setCloseAtEnd(true)
                .setCloseButton(true)
                .build();

        mAd.loadAd("ca-app-pub-7706302691807937/3465921207", new AdRequest.Builder().addNetworkExtrasBundle(SAAdMobVideoMediationAdapter.class, extras3).build());
    }

    public void playInterstitial (View view) {
        if (mInterstitialAd.isLoaded()) {
            mInterstitialAd.show();
        } else {
            Log.d("SADefaults/MoPub", "Interstitial not loaded yet");
        }
    }

    public void playVideo (View view) {
        if (mAd.isLoaded()) {
            mAd.show();
        } else {
            Log.d("SADefaults/MoPub", "Video Ad not loaded yet");
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        adView.destroy();
    }
}

File: app/src/main/java/tv/superawesome/demoapp/AdMobActivity.kt
package tv.superawesome.demoapp

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import com.google.android.gms.ads.*
import com.google.android.gms.ads.admanager.AdManagerInterstitialAd
import com.google.android.gms.ads.interstitial.InterstitialAd
import com.google.android.gms.ads.interstitial.InterstitialAdLoadCallback
import com.google.android.gms.ads.rewarded.RewardedAd
import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback
import tv.superawesome.plugins.publisher.admob.SAAdMobAdapter
import tv.superawesome.plugins.publisher.admob.SAAdMobExtras
import tv.superawesome.sdk.publisher.SAOrientation

class AdMobActivity : Activity() {
    private val tag = "SADefaults/AdMob"
    private lateinit var adView: AdView
    private var interstitialAd: InterstitialAd? = null
    private var rewardedAd: RewardedAd? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_admob)

        MobileAds.initialize(this) {
            Log.d(tag, "onInitializationComplete")
        }
        MobileAds.setRequestConfiguration(
            RequestConfiguration.Builder()
                .setTestDeviceIds(listOf("A86A81848A15ED8C5E27A1A72B263929")).build()
        )

        adView = findViewById(R.id.adView)
        requestBannerAd()
        findViewById<Button>(R.id.playInterstitialBtn)?.setOnClickListener {
            requestInterstitialAd()
        }
        findViewById<Button>(R.id.playVideoBtn)?.setOnClickListener {
            requestVideoAd()
        }
    }

    private fun requestBannerAd() {
        val bundle = SAAdMobExtras.extras()
            .setTestMode(false)
            .setParentalGate(false)
            .setTransparent(true)
            .build()

        adView.loadAd(
            AdRequest.Builder()
                .addNetworkExtrasBundle(SAAdMobAdapter::class.java, bundle)
                .build()
        )
    }

    private fun requestInterstitialAd() {
        val bundle = SAAdMobExtras.extras()
            .setTestMode(false)
            .setOrientation(SAOrientation.PORTRAIT)
            .setParentalGate(false)
            .build()

        AdManagerInterstitialAd.load(
            this,
            getString(R.string.admob_interstitial_ad_id),
            AdRequest.Builder()
                .addNetworkExtrasBundle(SAAdMobAdapter::class.java, bundle)
                .build(),
            object : InterstitialAdLoadCallback() {
                override fun onAdLoaded(ad: InterstitialAd) {
                    super.onAdLoaded(ad)
                    interstitialAd = ad
                    interstitialAd?.show(this@AdMobActivity)
                }

                override fun onAdFailedToLoad(error: LoadAdError) {
                    super.onAdFailedToLoad(error)
                }
            })
    }

    private fun requestVideoAd() {
        val bundle = SAAdMobExtras.extras()
            .setTestMode(false)
            .setParentalGate(false)
            .setOrientation(SAOrientation.PORTRAIT)
            .setSmallClick(true)
            .setCloseAtEnd(true)
            .setCloseButton(true)
            .build()

        RewardedAd.load(this,
            getString(R.string.admob_video_ad_id),
            AdRequest.Builder()
                .addNetworkExtrasBundle(SAAdMobAdapter::class.java, bundle)
                .build(),
            object : RewardedAdLoadCallback() {
                override fun onAdFailedToLoad(error: LoadAdError) {
                    super.onAdFailedToLoad(error)
                    Log.e(tag, error.message)
                }

                override fun onAdLoaded(ad: RewardedAd) {
                    super.onAdLoaded(ad)
                    rewardedAd = ad
                    rewardedAd?.show(this@AdMobActivity) {
                        Log.e(tag, it.amount.toString())
                    }
                }
            })
    }

    override fun onDestroy() {
        super.onDestroy()
        adView.destroy()
    }
}